### PR TITLE
Smart contract usage howto + bugfix

### PR DESCRIPTION
**Pull request by Zett98**
## Problem

We added rumintary smart-contract support to deku using lambda vm but we have no instructions how to use it.
## Solution

add the instructions.
Note: patch also includes a bugfix for contract invocation on lambda

### CODE REVIEW

File: src/lambda_vm/lambda_vm.ml [hunk 1]
```diff
@@ -15,7 +15,11 @@ let wrap compute =
   | exception Checks.Out_of_stack -> `Limits Runtime_limits_error.Out_of_stack
   | exception Gas.Out_of_gas -> `Limits Runtime_limits_error.Out_of_gas
 
-module Ir = Ir
+module Ir = struct
+  include Ir
+  let tup_arg_and_storage arg storage =
```
Comment (renatoalencar) [inline src/lambda_vm/lambda_vm.ml:20]: I'm not sure about this, I think we could either have type constraints for arg and storage, use the usual type constructors or have custom functions (like `pair`)  for creating this instead.

Comment (Zett98) [reply src/lambda_vm/lambda_vm.ml]: changed to have a module with Value constructors

File: src/core/contract_vm.ml [hunk 2]
```diff
@@ -72,7 +69,7 @@ module Lambda = struct
     type t = Raw_repr.Value.t [@@deriving yojson]
   end
   module Interpreter = struct
-    let error_to_string : Lambda_vm.Interpreter.error -> string = function
+    let pp_error : Lambda_vm.Interpreter.error -> string = function
```
Comment (renatoalencar) [inline src/core/contract_vm.ml:72]: `pp_` functions usually expect a `Format.formatter` value and return `unit`. I think `<something>_to_string` (or `of_string`) is better in this context.

File: src/lambda_vm/lambda_vm.ml [hunk 3]
```diff
@@ -15,7 +15,14 @@ let wrap compute =
   | exception Checks.Out_of_stack -> `Limits Runtime_limits_error.Out_of_stack
   | exception Gas.Out_of_gas -> `Limits Runtime_limits_error.Out_of_gas
 
-module Ir = Ir
+module Ir = struct
+  include Ir
+
+  module Value_syntax = struct
```
Comment (renatoalencar) [inline src/lambda_vm/lambda_vm.ml:21]: This looks great

**Review by renatoalencar — APPROVED** [review]
It looks great now 🚀

**Pull request merged**